shepherd/storage/storage.py

```diff
@@ -0,0 +1,53 @@
+import abc
+
+
+class Storage(metaclass=abc.ABCMeta):
+    """
+    An interface for services that provide access to job data in a remote storage
+    """
+
+    @abc.abstractmethod
+    def is_accessible(self) -> bool:
+        """
+        Check if the remote storage can be accessed
+        """
+
+    @abc.abstractmethod
+    def pull_job_data(self, job_id: str, target_directory: str):
+        """
+        Download job files from the remote storage to a local directory
```

Comment:
Missing newline after the description, missing dot at the end of the sentence... (multiple times)

Reply:
Missing args description